library/proc_macro/src/bridge/client.rs

```diff
@@ -49,16 +47,16 @@ macro_rules! define_handles {
             #[repr(C)]
             pub(crate) struct $oty {
                 handle: handle::Handle,
-                // Prevent Send and Sync impls
-                _marker: PhantomData<*mut ()>,
             }
 
+            impl !Send for $oty {}
+            impl !Sync for $oty {}
```

Comment:
I recently changed this in https://github.com/rust-lang/rust/pull/95308 because negative impls are unstable and rust-analyzer needs the bridge on stable. Using `PhantomData` instead of negative impls reduces the amount of patching necessary when importing as part of rust-analyzer.

Reply:
As an aside, I think the RA design is kind of broken and it should rather compile its own wrapper around the bridge, against the `libproc_macro.rlib` *shipped* with whatever Rust toolchain it's trying to support, using `RUSTC_BOOTSTRAP=1`, than copy-paste the `proc_macro` code.

As much as I dislike `RUSTC_BOOTSTRAP=1`, RA *must* interface with unstable Rust in a way privileged to the `rust-lang/rust` repo/build system, so we might as well not pretend it's not.

Reply:
> I recently changed this in https://github.com/rust-lang/rust/pull/95308 because negative impls are unstable and rust-analyzer needs the bridge on stable.

Interesting. Several of these types use `PhantomData<*mut ()>`, but `Closure` uses `PhantomData<*mut &'a mut ()>`. Can `Closure` instead use `PhantomData<*mut ()>`? At least that would remove the `'a`.

Reply:
I will drop this commit but add some more comments on these marker fields, explaining this additional info.

Reply:
One question about this: I didn't have to add a `#![feature(...)]` annotation in order to use `!Send`/`!Sync`. Why is that?